src/Lecture1.hs

```diff
@@ -123,4 +145,14 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: Int -> [Int] -> String
+lowerAndGreater n list = lowerAndGreater' n list 0 0
+                         where
+                            lowerAndGreater' _ [] bigger lower = show n ++
+                                " is greater than " ++ show lower ++
+                                " elements and lower than " ++
+                                show bigger ++ " elements"
+                            lowerAndGreater' m (x:xs) bigger lower
+                                           | m < x = lowerAndGreater' m xs (bigger+1) lower
+                                           | m > x = lowerAndGreater' m xs bigger (lower+1)
+                                           | otherwise = lowerAndGreater' m xs bigger lower
```

Comment:
I see that the number `m` that you pass in recursive calls doesn't change. You always pass the same number. In that case, you can remove this extra argument and use `n` from the function instead. It's possible because functions in `where` can see all the arguments of the top-level function 🔍